cmd/ingester/app/consumer/consumer.go

```diff
@@ -149,11 +150,17 @@ func (c *Consumer) handleMessages(pc sc.PartitionConsumer) {
 				defer msgProcessor.Close()
 			}
 
-			msgProcessor.Process(&saramaMessageWrapper{msg})
+			msgProcessor.Process(&saramaMessageWrapper{msg}, func(message processor.Message, e error) {
+				msgErrors <- e
```

Comment:
is the callback func() only called on error?  If it can be called on success, then we should check for err != nil.

nit: s/e/err/